test/throw.test.js

```diff
@@ -144,210 +141,209 @@ test('Should throw on duplicate decorator', t => {
   fastify.decorate('foo', fooObj)
   try {
     fastify.decorate('foo', fooObj)
-    t.fail()
+    t.assert.fail()
   } catch (e) {
-    t.pass()
+    t.assert.ok(true)
   }
 })
 
-test('Should not throw on duplicate decorator encapsulation', t => {
+test('Should not throw on duplicate decorator encapsulation', async (t) => {
   t.plan(1)
-
   const fastify = Fastify()
   const foo2Obj = {}
 
   fastify.decorate('foo2', foo2Obj)
 
   fastify.register(function (fastify, opts, done) {
-    t.doesNotThrow(() => {
+    t.assert.doesNotThrow(() => {
       fastify.decorate('foo2', foo2Obj)
     })
     done()
   })
 
-  fastify.ready()
+  await fastify.ready()
 })
 
-test('Should throw on duplicate request decorator', t => {
+test('Should throw on duplicate request decorator', async (t) => {
   t.plan(2)
 
   const fastify = Fastify()
 
   fastify.decorateRequest('foo', null)
   try {
     fastify.decorateRequest('foo', null)
-    t.fail()
+    t.assert.fail()
   } catch (e) {
-    t.equal(e.code, 'FST_ERR_DEC_ALREADY_PRESENT')
-    t.equal(e.message, 'The decorator \'foo\' has already been added!')
+    t.assert.strictEqual(e.code, 'FST_ERR_DEC_ALREADY_PRESENT')
+    t.assert.strictEqual(e.message, 'The decorator \'foo\' has already been added!')
```

Comment:
If someone in future would like to clean up the test, here we can use:

```js
t.assert.throws(()=>{fastify.decorateRequest('foo', null)}, {
    code: 'FST_ERR_DEC_ALREADY_PRESENT',
    message: 'The decorator \'foo\' has already been added!'
  })
```

https://nodejs.org/api/assert.html#assertthrowsfn-error-message

But out of scope of this PR